src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEBuildManager.cs

```diff
@@ -0,0 +1,568 @@
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.TempPE
+{
+    [Export(typeof(ITempPEBuildManager))]
+    [Export(ExportContractNames.Scopes.UnconfiguredProject, typeof(IProjectDynamicLoadComponent))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasicLanguageServiceWindowsForms)]
+    internal partial class TempPEBuildManager : UnconfiguredProjectHostBridge<IProjectVersionedValue<InputTuple>, TempPEBuildManager.DesignTimeInputsDelta, IProjectVersionedValue<TempPEBuildManager.DesignTimeInputsItem>>, ITempPEBuildManager, IProjectDynamicLoadComponent
+    {
+        private static readonly TimeSpan s_compilationWaitTime = TimeSpan.FromMilliseconds(500);
+
+        private readonly IUnconfiguredProjectCommonServices _unconfiguredProjectServices;
+        private readonly IActiveWorkspaceProjectContextHost _activeWorkspaceProjectContextHost;
+        private readonly IActiveConfiguredProjectSubscriptionService _projectSubscriptionService;
+        private readonly IFileSystem _fileSystem;
+        private readonly IProjectFaultHandlerService _projectFaultHandlerService;
+        private readonly VSBuildManager _buildManager;
+        private readonly ITempPECompiler _compiler;
+
+        [ImportingConstructor]
+        public TempPEBuildManager(IProjectThreadingService threadingService,
+                                  IUnconfiguredProjectCommonServices unconfiguredProjectServices,
+                                  IActiveWorkspaceProjectContextHost activeWorkspaceProjectContextHost,
+                                  IActiveConfiguredProjectSubscriptionService projectSubscriptionService,
+                                  ITempPECompiler compiler,
+                                  IFileSystem fileSystem,
+                                  IProjectFaultHandlerService projectFaultHandlerService,
+                                  VSBuildManager buildManager)
+             : base(threadingService.JoinableTaskContext)
+        {
+            _unconfiguredProjectServices = unconfiguredProjectServices;
+            _activeWorkspaceProjectContextHost = activeWorkspaceProjectContextHost;
+            _projectSubscriptionService = projectSubscriptionService;
+            _compiler = compiler;
+            _fileSystem = fileSystem;
+            _projectFaultHandlerService = projectFaultHandlerService;
+            _buildManager = buildManager;
+        }
+
+        public string[] GetTempPEMonikers()
+        {
+            Initialize();
+
+            return AppliedValue.Value.Inputs.ToArray();
+        }
+
+        /// <summary>
+        /// Called privately to actually fire the TempPE events and optionally recompile the TempPE library for the specified input
+        /// </summary>
+        private async Task FireTempPEDirtyAsync(string moniker, bool shouldCompile)
+        {
+            // Not using use the ThreadingService property because unit tests
+            await _unconfiguredProjectServices.ThreadingService.SwitchToUIThread();
```

Comment:
You mention doing this for unit tests, but I don't see any tests added in this PR. Is that change coming later?

Reply:
Yes, just trying to reduce the amount of code to review.